documentation/developer/adr/0002-record-design-stage.md

```diff
@@ -0,0 +1,61 @@
+# 2. `DesignStage` architectural decision
+
+Date: 2022-06-30
+Updated: 2022-10-24
+
+## Status
+
+Completed
+
+## Context
+
+At present we have a `design_stage` decorator which wraps methods in the `Reactor` base
+class. The `["run", "read", "mock"]` logic is presently handled by `Builder`s and `Solver`s.
+
+We want `Builder` to effectively only return a `Component` or tree of `Component`.
+
+We want `Builder` to be agnostic of the design problem and of `["run", "read", "mock"]` logic.
+
+We want `Builder` to have a minimum set of `Parameters`.
+
+`design_stage` presently only handles printing logic for entry and exit to an operation.
+
+Options:
+* Do nothing
+* Add a `DesignStage` class which handles the `["run", "read", "mock"]` logic for `Builder`
+
+* Should every `DesignStage` be a `Solver` where `["run", "read", "mock"]` logic is also
+handled?
+
+## Decision
+
+We add a `DesignStage` class to handles the  `["run", "read", "mock"]` logic for
+`Builder` only.
+
+We will trial the two options above taking e.g. the EUDEMO `PlasmaBuilder` as an example
+to inform our decision.
+
+#### 2022-10-24
+
+The `Designer` (previously called the `DesignStage`) is in charge of solving a minimal design problem.
+In its simplest form the output of the `Designer`s `execute` method is an input to the `Builder`.
+A `Designer` can have different run modes as described above but by default only needs a run method to function.
+
+With user inputs and the output of the `execute` method the `Builder` creates the CAD for a given `Component`.
+At the time of writing the `Builder` will output a `ComponentManager` containing the `Component` built by the `Builder`.
+We plan for a `Builder` to output a `Component` to allow flexibility in `ComponentManager` definition.
+The output of the `Builder` and possibly some output of the `Designer` would then be used to intialise a `ComponentManager`.
+
+Both a `Designer` and `Builder` need their own `ParameterFrame` to define the `Parameters` needed.
+
+A `ParameterFrame` should only be needed in the `Designer` or `Builder` and external functions should have the relevant `Parameter` values passed in directly.
```

Comment:
Should we use '_and/or_' here, because this makes it sound as if only one of them should have a `ParameterFrame` (at least to me)? 

> _A `ParameterFrame` should only be needed in the `Designer` or `Builder`_

And/or `Reactor`?

Reply:
unsure about what I've just written in [7bd94ca](https://github.com/Fusion-Power-Plant-Framework/bluemira/pull/1192/commits/7bd94ca8db3cf98c16599ddbbcafe869d7109113), let me know